gui/packages/ubuntupro/lib/pages/widgets/delayed_text_field.dart

```diff
@@ -0,0 +1,129 @@
+  const DelayedTextField({
+    this.autofocus = false,
+    this.enabled = true,
+    this.controller,
+    this.error,
+    this.errorText,
+    this.hintText,
+    this.inputFormatters,
+    this.label,
+    this.onChanged,
+    this.onSubmitted,
+    super.key,
+  });
+
+  final bool autofocus;
+  final TextEditingController? controller;
+  final bool enabled;
+  final Widget? error;
+  final String? errorText;
+  final String? hintText;
+  final List<TextInputFormatter>? inputFormatters;
+  final Widget? label;
+  final void Function(String)? onChanged;
+  final void Function(String)? onSubmitted;
+
+  @override
+  State<DelayedTextField> createState() => _DelayedTextField();
+}
+
+class _DelayedTextField extends State<DelayedTextField>
+    with SingleTickerProviderStateMixin {
+  late TimerNotifier debouncer;
+
+  bool showError = false;
+
+  @override
+  void initState() {
+    super.initState();
+    debouncer = TimerNotifier(vsync: this, duration: Durations.medium4);
+    debouncer.addListener(() {
+      showError = mounted && widget.error != null || widget.errorText != null;
+    });
+  }
+
+  @override
+  void dispose() {
+    debouncer.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ListenableBuilder(
+      listenable: debouncer,
+      builder: (context, _) {
+        return TextField(
+          controller: widget.controller,
+          autofocus: widget.autofocus,
+          inputFormatters: widget.inputFormatters,
+          onChanged: (value) {
+            widget.onChanged?.call(value);
+            debouncer.stop();
+            debouncer.resume();
+          },
+          onSubmitted: widget.onSubmitted,
+          decoration: InputDecoration(
+            error: showError ? widget.error : null,
+            errorText: showError ? widget.errorText : null,
+            label: widget.label,
+          ),
+        );
+      },
+    );
+  }
+}
+
+/// A [ChangeNotifier] that notifies when the provided duration elapses.
+class TimerNotifier extends ChangeNotifier {
+  TimerNotifier({required this.duration, required this.vsync})
+      : assert(duration > Duration.zero, 'Duration must be greater than zero') {
+    _ticker = vsync.createTicker(_onTick);
+  }
+
+  final TickerProvider vsync;
+  final Duration duration;
+
+  Ticker? _ticker;
+  Duration _elapsedTime = Duration.zero;
+
+  @override
+  void dispose() {
+    _ticker?.dispose();
+    super.dispose();
+  }
+
+  /// Starts the timer.
+  void start() {
+    _elapsedTime = Duration.zero;
+    _ticker?.start();
+  }
+
+  /// Stops the timer.
+  void stop() {
+    _ticker?.stop();
+    _elapsedTime = Duration.zero;
+  }
+
+  /// Resumes the timer from the last elapsed time.
+  void resume() {
+    _ticker?.start();
+  }
+
+  /// Callback executed on each tick.
+  void _onTick(Duration elapsed) {
+    _elapsedTime = elapsed;
+
+    if (_elapsedTime >= duration) {
+      _ticker?.stop();
+      notifyListeners();
```

Comment:
```suggestion
      scheduleMicrotask(notifyListeners());
```

Just a minor performance enhancement. Scheduling a microtask allow the current frame to finish rendering before notifying listeners, which prevents jittering and/or artifacts. Not an issue for this simple app, but it's a good practice overall.